core/stop/residual_norm_kernels.hpp

```diff
@@ -63,12 +63,39 @@ namespace residual_norm {
 }  // namespace residual_norm
 
 
+namespace implicit_residual_norm {
```

Comment:
nit: This namespace is unnecessary, as macros don't have namespaces. Might still keep it around to make it easier to understand which namespaces the kernels live in?

Reply:
Yes, for clarity, I think it is better to have the namespace.